File: app/src/main/java/com/example/catalogodeproductos/MainActivity.java
package com.example.catalogodeproductos;

import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;

import androidx.activity.OnBackPressedCallback;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;

public class MainActivity extends AppCompatActivity implements MenuFragment.OnOptionClickListener {

    private View navContainer;
    private boolean isMenuExpanded = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Toolbar
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        navContainer = findViewById(R.id.nav_view_container);

        // Botón atrás
        getOnBackPressedDispatcher().addCallback(this, new OnBackPressedCallback(true) {
            @Override
            public void handleOnBackPressed() {
                if (isMenuExpanded) {
                    toggleMenu();
                } else {
                    setEnabled(false);
                    getOnBackPressedDispatcher().onBackPressed();
                    setEnabled(true);
                }
            }
        });

        // Cargar perfil por defecto
        if (savedInstanceState == null) {
            onOptionClicked("profile");
        }
    }

    @Override
    public void onToggleMenu() {
        toggleMenu();
    }

    private void toggleMenu() {
        isMenuExpanded = !isMenuExpanded;
        
        ViewGroup.LayoutParams params = navContainer.getLayoutParams();
        // Cambiado a 180dp para que no se expanda tanto
        params.width = isMenuExpanded ? 
                (int) (180 * getResources().getDisplayMetrics().density) :
                (int) (70 * getResources().getDisplayMetrics().density);
        navContainer.setLayoutParams(params);

        MenuFragment menuFragment = (MenuFragment)
                getSupportFragmentManager().findFragmentById(R.id.menu_fragment_container);
        if (menuFragment != null) {
            menuFragment.setExpanded(isMenuExpanded);
        }
    }

    @Override
    public void onOptionClicked(String option) {
        Fragment fragment;

        switch (option) {
            case "photos":
                fragment = new PhotosFragment();
                break;
            case "video":
                fragment = new VideoFragment();
                break;
            case "web":
                fragment = new WebFragment();
                break;
            case "buttons":
                fragment = new ButtonsFragment();
                break;
            default:
                fragment = new ProfileFragment();
                break;
        }

        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.content_fragment_container, fragment)
                .commit();

        MenuFragment menuFragment = (MenuFragment)
                getSupportFragmentManager().findFragmentById(R.id.menu_fragment_container);

        if (menuFragment != null) {
            menuFragment.highlightOption(option);
        }
        
        // El menú se queda en su estado actual (abierto o cerrado) al cambiar de módulo
    }
}

File: app/src/main/java/com/example/catalogodeproductos/MainActivity.kt
package com.example.catalogodeproductos

import android.os.Bundle
import androidx.activity.OnBackPressedCallback
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.Fragment
import androidx.fragment.app.commit

class MainActivity : AppCompatActivity(), MenuFragment.OnOptionClickListener {

    private lateinit var drawerLayout: DrawerLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Configuración de la Toolbar
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        // Configuración del Drawer (Menú lateral oculto)
        drawerLayout = findViewById(R.id.drawer_layout)
        val toggle = ActionBarDrawerToggle(
            this, drawerLayout, toolbar,
            R.string.navigation_drawer_open, R.string.navigation_drawer_close
        )
        drawerLayout.addDrawerListener(toggle)
        toggle.syncState()

        // Migración de onBackPressed a OnBackPressedDispatcher
        onBackPressedDispatcher.addCallback(this, object : OnBackPressedCallback(true) {
            override fun handleOnBackPressed() {
                if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
                    drawerLayout.closeDrawer(GravityCompat.START)
                } else {
                    // Si el menú no está abierto, permitimos el comportamiento normal (salir de la app)
                    isEnabled = false
                    onBackPressedDispatcher.onBackPressed()
                    isEnabled = true
                }
            }
        })

        // Cargar el perfil por defecto al iniciar
        if (savedInstanceState == null) {
            onOptionClicked("profile")
        }
    }

    // Método que se activa al elegir una opción del menú o del footer
    override fun onOptionClicked(option: String) {
        val fragment: Fragment = when (option) {
            "photos" -> PhotosFragment()
            "video" -> VideoFragment()
            "web" -> WebFragment()
            "buttons" -> ButtonsFragment()
            else -> ProfileFragment()
        }

        // Reemplazar el fragmento actual
        supportFragmentManager.commit {
            replace(R.id.content_fragment_container, fragment)
            setReorderingAllowed(true)
        }

        // Cerrar el menú lateral después de seleccionar
        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
            drawerLayout.closeDrawer(GravityCompat.START)
        }
    }
}
